File: homework-assignment-0/word_count.py
from pyspark import SparkContext
import os

os.environ['PYSPARK_PYTHON'] = '/opt/homebrew/opt/python@3.10/bin/python3.10'
os.environ['PYSPARK_DRIVER_'] = '/opt/homebrew/opt/python@3.10/bin/python3.10'

sc = SparkContext('local[*]', 'wordCount')

input_file_path = './text.txt'
textRDD = sc.textFile(input_file_path)

word_counts = textRDD.flatMap(lambda line: line.split(' ')) \
    .map(lambda word: (word, 1)) \
    .reduceByKey(lambda count1, count2: count1 + count2) \
    .collect()

for word, count in word_counts:
    print(word, count)

File: homework-assignment-0/scala-hw-0/src/main/scala/WordCount.scala
package hw_0

import org.apache.spark.{SparkConf, SparkContext}

object WordCount {

  def main(arg: Array[String]): Unit = {

    val conf = new SparkConf().setAppName("word-count").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val datasetRDD = sc.textFile("../text.txt")
    val counts = datasetRDD.flatMap(line => line.split(" ")).map(word => (word, 1)).reduceByKey(_ + _)
    counts.foreach(println)
  }
}